src/Lykke.Service.FixGateway.Services/LimitOrderNotificationsListener.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.Service.FixGateway.Core.Services;
+using Lykke.Service.FixGateway.Core.Settings.ServiceSettings;
+using Lykke.Service.FixGateway.Services.DTO.MatchingEngine;
+using QuickFix.Fields;
+using QuickFix.FIX44;
+using ILog = Common.Log.ILog;
+using Message = QuickFix.Message;
+using OrderStatus = Lykke.Service.FixGateway.Services.DTO.MatchingEngine.OrderStatus;
+
+namespace Lykke.Service.FixGateway.Services
+{
+    [UsedImplicitly]
+    public sealed class LimitOrderNotificationsListener : IDisposable
+    {
+        private readonly IClientOrderIdProvider _clientOrderIdProvider;
+        private readonly SessionState _sessionState;
+        private readonly IMapper _mapper;
+        private readonly IFixMessagesSender _fixMessagesSender;
+        private readonly ILog _log;
+        private readonly string _clientId;
```

Comment:
I'm not sure it is a good practice to have dedicated listener for each client.

Reply:
There will only one client per the microservice instance. So it is not a problem for now.